### ISSUE TITLE
Title overflows course card

### ISSUE THREAD

**Issue by pdpinch** [mitodl/ocw-hugo-themes#746]
### Steps to Reproduce

See https://ocw.mit.edu/

![image (20)](https://user-images.githubusercontent.com/430126/173129382-c8605509-b8e4-43ef-8955-8062ae086172.png)

**Comment by pdpinch** [comment]
Until we can find a design solution,  can we try clipping the title after 4 lines?

**Comment by gumaerc** [comment]
@pdpinch I investigated a number of different ways to do this today.  Apparently truncating a single line of text is straightforward and built into the CSS spec as `text-overflow: ellipsis`.  Line clamp for multiline text is unfortunately not so simple as is detailed [here](https://css-tricks.com/line-clampin/).  I tried a number of the solutions detailed in this blog post.  The first solution recommended didn't work properly for me and it also uses webkit specific rules which might not be great for something like OCW.  There are Javascript solutions as well like [Clamp.js](https://github.com/josephschmitt/Clamp.js/) but that would add a 3rd party dependency which isn't great for a one-off thing like this.  On his site he also talks about how the performance of Clamp.js isn't great and it shouldn't be used in production systems.

This led me to just try a modified version of one of the solutions he suggests at the end, simply using `max-height` calculated based on the line height of the text and a maximum number of lines:

```
    .course-card-title {
      --max-lines: 3;
      --line-height: 1.5rem;
      position: relative;
      max-height: calc(var(--line-height) * var(--max-lines));
      overflow: hidden;
    }
```

This results in something like this (there are a lot more "test" strings than shown in the HTML):

![chrome_2022-06-13_18-07-22](https://user-images.githubusercontent.com/12089658/173453253-2d6cbd2f-0acf-4e44-9ca0-7f93f4f664c8.jpg)

I was having trouble getting the ellipsis to show properly because it has to be added using absolute positioning and padding which can be kind of janky to do.  If simply clamping the number of lines is all we need, then this is all we need to do.  If we want an ellipsis though, I could spend a bit more time to refine that.
